jsonrpc/types.go

```diff
@@ -112,28 +112,65 @@ func toTransaction(
 	return res
 }
 
+type header struct {
+	ParentHash      types.Hash  `json:"parentHash"`
+	Sha3Uncles      types.Hash  `json:"sha3Uncles"`
+	Miner           argBytes    `json:"miner"`
+	StateRoot       types.Hash  `json:"stateRoot"`
+	TxRoot          types.Hash  `json:"transactionsRoot"`
+	ReceiptsRoot    types.Hash  `json:"receiptsRoot"`
+	LogsBloom       types.Bloom `json:"logsBloom"`
+	Difficulty      argUint64   `json:"difficulty"`
+	TotalDifficulty argUint64   `json:"totalDifficulty"`
+	Number          argUint64   `json:"number"`
+	GasLimit        argUint64   `json:"gasLimit"`
+	GasUsed         argUint64   `json:"gasUsed"`
+	Timestamp       argUint64   `json:"timestamp"`
+	ExtraData       argBytes    `json:"extraData"`
+	MixHash         types.Hash  `json:"mixHash"`
+	Nonce           types.Nonce `json:"nonce"`
+	Hash            types.Hash  `json:"hash"`
+	BaseFee         argUint64   `json:"baseFeePerGas,omitempty"`
+}
+
+type accessListResult struct {
+	Accesslist types.TxAccessList `json:"accessList"`
+	Error      error              `json:"error,omitempty"`
+	GasUsed    argUint64          `json:"gasUsed"`
+}
+
 type block struct {
-	ParentHash      types.Hash          `json:"parentHash"`
-	Sha3Uncles      types.Hash          `json:"sha3Uncles"`
-	Miner           argBytes            `json:"miner"`
-	StateRoot       types.Hash          `json:"stateRoot"`
-	TxRoot          types.Hash          `json:"transactionsRoot"`
-	ReceiptsRoot    types.Hash          `json:"receiptsRoot"`
-	LogsBloom       types.Bloom         `json:"logsBloom"`
-	Difficulty      argUint64           `json:"difficulty"`
-	TotalDifficulty argUint64           `json:"totalDifficulty"`
-	Size            argUint64           `json:"size"`
-	Number          argUint64           `json:"number"`
-	GasLimit        argUint64           `json:"gasLimit"`
-	GasUsed         argUint64           `json:"gasUsed"`
-	Timestamp       argUint64           `json:"timestamp"`
-	ExtraData       argBytes            `json:"extraData"`
-	MixHash         types.Hash          `json:"mixHash"`
-	Nonce           types.Nonce         `json:"nonce"`
-	Hash            types.Hash          `json:"hash"`
-	Transactions    []transactionOrHash `json:"transactions"`
-	Uncles          []types.Hash        `json:"uncles"`
-	BaseFee         argUint64           `json:"baseFeePerGas,omitempty"`
+	header
+	Size         argUint64           `json:"size"`
+	Transactions []transactionOrHash `json:"transactions"`
+	Uncles       []types.Hash        `json:"uncles"`
+}
+
+func toAccessList(aList map[types.Address]map[types.Hash]struct{}) types.TxAccessList {
```

Comment:
I would move this function to the `runtime.AccessList` object, and just call it `ConvertToTypes` or something like that.

Reply:
It can, but one of the previous CR comments: "IMO it is not worth it importing the "github.com/0xPolygon/polygon-edge/state/runtime" package, just because of AccessList struct. Though not sure is there some more elegant approach..."

Reply:
But `runtime` package already imports `types` package. You will only be moving this function to the `runtime.AccessList` struct. I think it is more clean this way.

Reply:
Ok. I thought you wanted to get rid of "map[types.Address]map[types.Hash]struct{}"